### PR TITLE
ref(dateTime): Use updated formatting rules

### PR DESCRIPTION
**Pull request by vuluongj20**
Refactor `DateTime` to reflect new formatting rules recently adopted by the design team and address inconsistencies between different configurations.

**The default formatting** (when no modifying prop like `seconds` and `timeZone` is present) has changed:
- **year**: only shown if the date is not in the current calendar year, e.g. `Jun 14` (2022), `Jan 1` (2022), `Dec 31, 2021`
- **seconds**: not shown by default
- **time zone**: not shown by default, unless the date object is in UTC

For example: before, by default we would get something like "Jun 14, 2022 4:00 PM PST", but now we get "Jun 14, 4:00 PM"

**The props** that `DateTime` accepts has also changed:
 - **Remove `timeAndDate`** (which forces the format`MMM DD, HH:mm`/`MMM DD, LT`), since the default formatting is now already very similar to it, so there's no need to specify it anymore
 - **Remove `shortDate`**, since it's only used in one place (Account Subscriptions in Settings), and the default format works well enough there
 - **Add a `year` prop** to override the default year formatting (see above). This is useful for when we want to always show the year.
 - **Add a `timeZone` prop** to override the default time zone formatting (see above).

Requires https://github.com/getsentry/getsentry/pull/7653

### CODE REVIEW

File: static/app/components/events/interfaces/spans/spanDetail.tsx [hunk 1]
```diff
@@ -387,7 +387,7 @@ class SpanDetail extends Component<Props, State> {
                   fixed: 'Mar 16, 2020 9:10:12 AM UTC',
                   value: (
                     <Fragment>
-                      <DateTime date={startTimestamp * 1000} />
+                      <DateTime date={startTimestamp * 1000} year seconds timeZone />
```
Comment (vuluongj20) [inline static/app/components/events/interfaces/spans/spanDetail.tsx:390]: We still want to show the full time string here, since it's a data field.
image

File: static/app/utils/discover/fieldRenderers.tsx [hunk 2]
```diff
@@ -123,7 +123,7 @@ export const FIELD_FORMATTERS: FieldFormatters = {
       <Container>
         {data[field]
           ? getDynamicText({
-              value: <FieldDateTime date={data[field]} />,
+              value: <FieldDateTime date={data[field]} year seconds timeZone />,
```
Comment (vuluongj20) [inline static/app/utils/discover/fieldRenderers.tsx:126]: Full time here also:
image

File: static/app/views/alerts/rules/metric/details/metricHistory.tsx [hunk 3]
```diff
@@ -114,6 +114,9 @@ function MetricAlertActivity({organization, incident}: MetricAlertActivityProps)
           value: incident.dateCreated,
           fixed: 'Mar 4, 2022 10:44:13 AM UTC',
         })}
+        year
+        seconds
+        timeZone
```
Comment (vuluongj20) [inline static/app/views/alerts/rules/metric/details/metricHistory.tsx:119]: Full time
image

File: static/app/views/settings/account/accountSubscriptions.tsx [hunk 4]
```diff
@@ -123,7 +123,6 @@ class AccountSubscriptions extends AsyncView<AsyncView['props'], State> {
                                   email: subscription.email,
                                   date: (
                                     <DateTime
-                                      shortDate
```
Comment (vuluongj20) [inline static/app/views/settings/account/accountSubscriptions.tsx:126]: Before
Screen Shot 2022-06-14 at 3 16 08 PM

After
image

File: tests/js/spec/utils/discover/fieldRenderer.spec.jsx [hunk 5]
```diff
@@ -133,7 +133,7 @@ describe('getFieldRenderer', function () {
     const renderer = getFieldRenderer('timestamp.to_day', {'timestamp.to_day': 'date'});
     const wrapper = mountWithTheme(renderer(data, {location, organization}));
     const text = wrapper.find('Container');
-    expect(text.text()).toEqual('September 5, 2021');
+    expect(text.text()).toEqual('Sep 5, 2021');
```
Comment (vuluongj20) [inline tests/js/spec/utils/discover/fieldRenderer.spec.jsx:136]: This was an edge case within our formatting system. We always want to use the short, 3-letter month format, never the full one. This rule is now enforced in `Datetime`.

File: static/app/components/dateTime.tsx [hunk 6]
```diff
@@ -4,81 +4,119 @@ import momentTimezone from 'moment-timezone';
...
+    format ??
+    getFormat({
+      dateOnly,
+      timeOnly,
+      // If the year prop is defined, then use it. Otherwise only show the year if `date`
+      // is in the current year.
+      year: year ?? moment().year() !== moment(date).year(),
+      // If timeZone is defined, use it. Otherwise only show the time zone if we're using
+      // UTC time.
+      timeZone: timeZone ?? utc,
```
Comment (vuluongj20) [inline static/app/components/dateTime.tsx:116]: The default assumption is that displayed dates are in the user's time zone. If the date is based on UTC, then we should explicitly show `UTC` as the time zone to avoid confusion.

**Comment by scttcper** [comment]
If this is the group events page, i'm not sure we can remove seconds here

jest/group-events-renders
image

**Comment by vuluongj20** [comment]
> If this is the group events page, i'm not sure we can remove seconds here
> 
> jest/group-events-renders image

@scttcper True, fixed in https://github.com/getsentry/sentry/pull/35666/commits/223b730a1271913298a4950e063f2fac72afe02c

**Comment by scttcper** [comment]
i think these dates on events page should keep their timezone info
image

**Comment by vuluongj20** [comment]
> i think these dates on events page should keep their timezone info image

@scttcper Hmm they look the same before and after for me. I think those dates use a custom formatting string (?), in which case they won't be affected:
image

**Review by scttcper — APPROVED** [review]
most of the dates in issues/alerts look good, there's probably still 100 more places
